internal/pkg/aws/acm/acm.go

```diff
@@ -66,23 +69,35 @@ func (a *ACM) ValidateCertAliases(aliases []string, certs []string) error {
 	}
 	for _, alias := range aliases {
 		if !validAliases[alias] {
-			return fmt.Errorf("%s is not a valid domain against %s", alias, strings.Join(certs, ","))
+			return &errInValidAliasAgainstCert{
+				certs:         certs,
+				alias:         alias,
+				domainsOfCert: domainsOfCert,
+			}
 		}
 	}
 	return nil
 }
 
-func (a *ACM) findValidAliasesAgainstCert(ctx context.Context, aliases []string, cert string) ([]string, error) {
+func (a *ACM) validDomainsOfCert(ctx context.Context, cert string) ([]string, error) {
 	resp, err := a.client.DescribeCertificateWithContext(ctx, &acm.DescribeCertificateInput{
 		CertificateArn: aws.String(cert),
 	})
 	if err != nil {
 		return nil, fmt.Errorf("describe certificate %s: %w", cert, err)
 	}
-	domainSet := make(map[string]bool)
-	domainSet[aws.StringValue(resp.Certificate.DomainName)] = true
+	var domainsOfCert []string
+	domainsOfCert = append(domainsOfCert, aws.StringValue(resp.Certificate.DomainName))
 	for _, san := range resp.Certificate.SubjectAlternativeNames {
-		domainSet[aws.StringValue(san)] = true
+		domainsOfCert = append(domainsOfCert, aws.StringValue(san))
+	}
+	return domainsOfCert, err
+}
+
+func filterValidAliases(domains []string, aliases []string) []string {
+	domainSet := make(map[string]bool)
```

Comment:
Since we know the length in advance, there is a chance for a bit of optimization:
```suggestion
	domainSet := make(map[string]bool, len(domains))
```
When we allocate the map with a initial size, we save the computation cost of resizing the map for the first `len(domains)` items.